CMakeLists.txt

```diff
@@ -92,32 +91,33 @@ add_executable(WCSim WCSim.cc ${sources} ${headers})
 target_link_libraries(WCSim ${Geant4_LIBRARIES} ${ROOT_LIBRARIES} WCSimRoot Tree)  #add profiler to use gperftools
 
 
-
 #----------------------------------------------------------------------------
-# Copy all scripts to the build directory, i.e. the directory in which we
+# Copy required files to the build directory, i.e. the directory in which we
 # build WCSim. This is so that we can run the executable directly because it
-# relies on these scripts being in the current working directory.
+# relies on these files being in the current working directory and then so
+# that we can directly analyze the produced ROOT files, too.
 #
-set(WCSIM_SCRIPTS
-  macros/jobOptions.mac
-  WCSim.mac
-  macros/daq.mac
-  macros/visOGLSX.mac
-  macros/visOGLQT.mac
-  macros/visRayTracer.mac
-  macros/mPMT_nuPrism1.mac
-  macros/mPMT_nuPrism2.mac
-  macros/tuning_parameters.mac
-  macros/NuPRISM.mac
-  )
-
-foreach(_script ${WCSIM_SCRIPTS})
-  configure_file(
-    ${PROJECT_SOURCE_DIR}/${_script}
-    ${PROJECT_BINARY_DIR}/${_script}
-    COPYONLY
+if (NOT (PROJECT_SOURCE_DIR STREQUAL PROJECT_BINARY_DIR))
+  file(COPY ${PROJECT_SOURCE_DIR}/mPMT-configfiles DESTINATION ${PROJECT_BINARY_DIR})
+  file(COPY ${PROJECT_SOURCE_DIR}/macros DESTINATION ${PROJECT_BINARY_DIR})
```

Comment:
Can you explain why this change is necessary? CMake `file(COPY ...)` lines cause problems on our cluster. This makes building many software painful (and the administrators can't / won't make changes to fix it) so I'd rather not make WCSim also painful to build unless it is really necessary.

Reply:
When the CMake building method is used in a separate directory, one needs to copy two directories "`mPMT-configfiles`" and "`macros`" (which are then needed at run-time).
In the old "`CMakeLists.txt `" file, there was only a partial explicit list of files from "`macros`" to be copied.
Of course, one can again make a full explicit list of all currently existing files and put it into the "`CMakeLists.txt `" file.
However, when people add or remove files from these directories, nobody really remembers to update this fixed list in this file.
As a result, one either has missing files (when some files have been added) or the CMake dies (when some files have been deleted).

So, it seems to me that the best idea is to make sure that always entire directories are copied automatically (using `file(COPY ...)` lines).

Could you, please, tell me what problems you face with these `file(COPY ...)` lines?

If they do not work for you, could you, please, try an equivalent form which is also expected to work:
```
  file(COPY ./mPMT-configfiles DESTINATION ./ )
  file(COPY ./macros DESTINATION ./ )
```

Reply:
The problem we have with these commands is due to how the filesystem quota works on our cluster. When creating files in the shared project space they must belong to the correct group, not just the user creating the file. Usually the filesystem handles this and we don't need to worry, but some CMake commands create files without the correct group ownership for some reason.
Anyway, after some investigating, this command can be made to work in our situation by also specifying `USE_SOURCE_PERMISSIONS`, i.e.:
```
  file(COPY ${PROJECT_SOURCE_DIR}/mPMT-configfiles DESTINATION ${PROJECT_BINARY_DIR} USE_SOURCE_PERMISSIONS)
  file(COPY ${PROJECT_SOURCE_DIR}/macros DESTINATION ${PROJECT_BINARY_DIR} USE_SOURCE_PERMISSIONS)
```
Could you update the commands to include this, then I think it all should be fine.

Reply:
pull request updated (see below)